streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java

```diff
@@ -77,10 +77,15 @@
                     DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG + " appropriately.",
                     deserializationException);
             } else {
-                sourceNode.nodeMetrics.sourceNodeSkippedDueToDeserializationError.record();
+                log.warn(
```

Comment:
I think you can still use the log4j format here, e.g.

```
"Skipping record due to deserialization error. topic={} partition={} offset={}", rawRecord.topic(), rawRecord.partition(), rawRecord.offset(), deserializationException
```

With four parameters, the last one is auto interpreted as the exception; maybe we can validate if this is the case.

Reply:
Confirmed, switching to the variant you mentioned still prints:
```
[2018-04-04 18:23:20,310] WARN task [0_1] Skipping record due to deserialization error. topic=[topic1] partition=[1] offset=[1] (org.apache.kafka.streams.processor.internals.RecordDeserializer:80)
org.apache.kafka.common.errors.SerializationException: Size of data received by IntegerDeserializer is not 4
```